snmp/tests/test_check.py

```diff
@@ -139,8 +139,34 @@ def test_scalar(aggregator):
 
     # Test service check
     aggregator.assert_service_check("snmp.can_check", status=SnmpCheck.OK, tags=common.CHECK_TAGS, at_least=1)
+    aggregator.assert_metric(
+        'datadog.snmp.submitted_metrics', value=99, metric_type=aggregator.GAUGE, count=1, tags=common.CHECK_TAGS
+    )
 
-    common.assert_common_metrics(aggregator)
+    common.assert_common_metrics(aggregator, tags=common.CHECK_TAGS)
+    aggregator.all_metrics_asserted()
+
+
+def test_submitted_metrics_count(aggregator):
+    instance = common.generate_instance_config(common.SCALAR_OBJECTS)
+    check = common.create_check(instance)
+    check.check(instance)
+
+    for metric in common.SCALAR_OBJECTS:
+        metric_name = "snmp." + (metric.get('name') or metric.get('symbol'))
+        aggregator.assert_metric(metric_name, tags=common.CHECK_TAGS, count=1)
+
+    aggregator.assert_metric('snmp.sysUpTimeInstance', count=1)
+    aggregator.assert_metric(
+        'datadog.snmp.submitted_metrics', value=5.0, metric_type=aggregator.GAUGE, count=1, tags=common.CHECK_TAGS
```

Comment:
should `value` be `len(common.SCALAR_OBJECTS)` ?

Reply:
It's more precisely `len(common.SCALAR_OBJECTS) + 1`, the +1 is `snmp.sysUpTimeInstance`.

Reply:
I'm thinking of a potential error not caught: we add a value to `SCALAR_OBJECTS`, and for some reason it is not considered on the `submitted_metrics` => this test would still pass, which would be a false positive